cityofaustin/atd-moped
# Drop unused moped_project columns and financial tables

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/10954
- https://github.com/cityofaustin/atd-data-tech/issues/11721

## Testing
**URL to test:** Local


**Steps to test:**
First, you're testing to make sure the app behaves normally.

1. Start the hasura cluster
2. Verify the project list loads
3. Verify the project summary and all other tabs load normally
4. Try updating a few fields on the project summary tab - make sure your changes are landing in the activity log
5. Create a new project and verify that it loads - double check its activity log

There is also a migration which copies the old `work_assignment_id` column into new `moped_proj_contract` records. To test this, you'll need to start the cluster with a copy of production. After applying migrations, you can:

1. Connect to the Moped DB with a SQL client and run

```sql
select * from moped_proj_contract order by id desc;
```

2. Observe that the first 33 records have null values except for the `work_assignment_id` column. Those are the newly created contract records.

3. These inserts will **not** show up in the activity log, because the event mutation is failing because of an error with `updated_by_user_id` . You can inspect the failing mutations by inspecting the `request` and `response` columns returned from:

```sql
select * from hdb_catalog.event_invocation_logs where trigger_name = 'activity_log_moped_proj_contract' order by created_at desc;
```

Here's an example error:

```json
{
  "data": {
    "body": "{\"errors\":[{\"extensions\":{\"code\":\"constraint-violation\",\"path\":\"$.selectionSet.insert_moped_activity_log_one.args.object\"},\"message\":\"Foreign key violation. insert or update on table \\\"moped_activity_log\\\" violates foreign key constraint \\\"moped_activity_log_updated_by_user_id_fkey\\\"\"}]}",
    "headers": [
      { "name": "Transfer-Encoding", "value": "chunked" },
      { "name": "Date", "value": "Tue, 14 Mar 2023 16:05:30 GMT" },
      { "name": "Server", "value": "Warp/3.3.23" },
      { "name": "Content-Type", "value": "application/json; charset=utf-8" }
    ],
    "status": 200
  },
  "type": "webhook_response",
  "version": "2"
}
```

As you can see, we [use a fallback user ID](https://github.com/cityofaustin/atd-moped/blob/abd890d1f8f7af826d8124c19e01f487bb1cf610/moped-database/metadata/tables.yaml#L1903) value of `1` - this value works locally (it's our default JD user), but we do not have a user ID value of `1` in production. 

I'm going to think about that problem but I think it's ok for now that these activity events will be missing from the log. Right?

---
#### Ship list
